core/src/main/java/com/linecorp/armeria/server/Http2ServerConnectionHandler.java

```diff
@@ -16,26 +16,32 @@
 
 package com.linecorp.armeria.server;
 
+import java.time.Duration;
+
 import com.linecorp.armeria.internal.AbstractHttp2ConnectionHandler;
 
 import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandlerContext;
 import io.netty.handler.codec.http2.Http2ConnectionDecoder;
 import io.netty.handler.codec.http2.Http2ConnectionEncoder;
 import io.netty.handler.codec.http2.Http2Settings;
+import io.netty.handler.timeout.IdleStateEvent;
 
 final class Http2ServerConnectionHandler extends AbstractHttp2ConnectionHandler {
 
     private final GracefulShutdownSupport gracefulShutdownSupport;
     private final Http2RequestDecoder requestDecoder;
+    private final Http2KeepAliveHandler keepAlive;
 
     Http2ServerConnectionHandler(Http2ConnectionDecoder decoder, Http2ConnectionEncoder encoder,
                                  Http2Settings initialSettings, Channel channel, ServerConfig config,
                                  GracefulShutdownSupport gracefulShutdownSupport, String scheme) {
 
         super(decoder, encoder, initialSettings);
-
         this.gracefulShutdownSupport = gracefulShutdownSupport;
-        requestDecoder = new Http2RequestDecoder(config, channel, encoder(), scheme);
+        keepAlive = new Http2KeepAliveHandler(channel, encoder().frameWriter(),
+                                              Duration.ofSeconds(1).toNanos());
```

Comment:
This seems really short. How about 10 seconds? How about making it configurable via `ServerBuilder` (and `ClientFactoryBuilder` if you're gonna implement the client side as well), preferably with a new `Flags` property?

Reply:
This was hardcoded for local testing. I will use `Flags` property.